packages/devtools_app/lib/src/bloc/instance_viewer/eval.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A few utilities related to evaluating dart code
+library eval;
+
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../../eval_on_dart_library.dart';
+import '../../globals.dart';
+import '../../vm_service_wrapper.dart';
+
+/// Exposes the current VmServiceWrapper.
+/// By listening to this provider instead of directly accessing `serviceManager.service`,
+/// this ensures that providers reload properly when the devtool is connected
+/// to a different application.
+final serviceProvider = StreamProvider<VmServiceWrapper>((ref) async* {
+  yield serviceManager.service;
+  yield* serviceManager.onConnectionAvailable;
+});
+
+/// An [EvalOnDartLibrary] that has access to no specific library in particular
+///
+/// Not suitable to be used when evaluating third-party objects, as it would
+/// otherwise not be possible to read private properties.
+final evalProvider = libraryEvalProvider('dart:io');
+
+final blocEvalProvider = libraryEvalProvider('package:bloc/src/bloc_observer.dart');
+/// An [EvalOnDartLibrary] that has access to `provider`
+final providerEvalProvider =
```

Comment:
do we need this?

Reply:
the blocEvalProvider gets used in instance_providers.dart

Reply:
we can change it to create an instance of evalOnDart directly in the file instead of importing it

Reply:
Sorry I meant the `providerEvalProvider`.